test/e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,167 @@
+
+	operatorDeploymentRetryInterval = time.Second * 5
+	operatorDeploymentTimeout       = time.Second * 120
+)
+
+func TestMain(m *testing.M) {
+	if isGinkgoRunner() {
+		err := bootstrapOperatorSDK()
+		if err != nil {
+			Logf("Failed to bootsrap operator SDK: %s", err.Error())
+			os.Exit(1)
+		}
+	}
+
+	framework.MainEntry(m)
+}
+
+func TestBuildRun(t *testing.T) {
+	testingT = t
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "E2E Suite")
+}
+
+var _ = SynchronizedBeforeSuite(func() []byte {
+	var err error
+
+	if isGinkgoRunner() {
+		Logf("Starting local operator")
+		operatorCmd, err = startLocalOperator()
+		Expect(err).ToNot(HaveOccurred(), "Failed to start local operator")
+	}
+
+	// add schemes to Operator SDK framework
+	Logf("Adding schemes to Operator SDK framework")
+	err = populateFrameworkScheme()
+	Expect(err).ToNot(HaveOccurred(), "Failed to add schemes to Operator SDK framework")
+
+	// setup the Operator SDK testing global context
+	globalCtx = framework.NewTestCtx(testingT)
+
+	// initialize cluster resources
+	Logf("Initializing cluster resources")
+	err = globalCtx.InitializeClusterResources(cleanupOptions(globalCtx))
+	Expect(err).ToNot(HaveOccurred(), "unable to initialize cluster resources")
+
+	// get the namespace
+	Logf("Getting the namespace")
+	namespace, err := globalCtx.GetNamespace()
+	Expect(err).ToNot(HaveOccurred(), "unable to obtain namespace")
+	Logf("Namespace is %s", namespace)
+
+	// wait for operator deployment
+	Logf("Waiting for operator deployment")
+	f := framework.Global
+	err = e2eutil.WaitForOperatorDeployment(
+		testingT,
+		f.KubeClient,
+		namespace,
+		"build-operator",
+		1,
+		operatorDeploymentRetryInterval,
+		operatorDeploymentTimeout,
+	)
+	Expect(err).ToNot(HaveOccurred(), "error on waiting for operator deployment")
+
+	// create the pipeline service account
+	Logf("Creating the pipeline service account")
+	createPipelineServiceAccount(globalCtx, f, namespace)
+
+	// create the container registry secret
+	Logf("Creating the container registry secret")
+	createContainerRegistrySecret(globalCtx, f, namespace)
+
+	// create cluster build strategies
+	Logf("Creating cluster build strategies")
+	for _, clusterBuildStrategy := range clusterBuildStrategies {
+		Logf("Creating cluster build strategy %s", clusterBuildStrategy)
+		cbs, err := clusterBuildStrategyTestData(clusterBuildStrategy)
+		Expect(err).ToNot(HaveOccurred())
+		cbs.SetNamespace(namespace)
+
+		createClusterBuildStrategy(globalCtx, f, cbs)
+	}
+
+	// create namespace build strategies
+	Logf("Creating namespace build strategies")
+	for _, namespaceBuildStrategy := range namespaceBuildStrategies {
+		Logf("Creating namespace build strategy %s", namespaceBuildStrategy)
+		nbs, err := buildStrategyTestData(namespace, namespaceBuildStrategy)
+		Expect(err).ToNot(HaveOccurred())
+
+		createNamespacedBuildStrategy(globalCtx, f, nbs)
+	}
+
+	return nil
+}, func(data []byte) {
+	if getGinkgoNode() > 1 {
```

Comment:
is this if conditional needed?

Reply:
No, it is not, it just prevents re-doing the code within the `if` clause for Ginkgo node 1 as it did it already in the previous function that only node 1 executes. Re-doing it had no harm based on my experience. So, I have no strong opinion here. If you insist, I remove the if-check.

Reply:
Agreed with @qu1queee; will remove the `if`.

Reply:
Done, see https://github.com/redhat-developer/build/pull/163/commits/7ee9f0765f5f92753d685a0c5ca7efed0be5665d#diff-61994e503eedf336006b7cb84a0fa60aR157